test/EVMHost.cpp

```diff
@@ -250,23 +250,48 @@ evmc::result EVMHost::call(evmc_message const& _message) noexcept
 
 	if (message.kind == EVMC_CREATE)
 	{
-		// TODO this is not the right formula
 		// TODO is the nonce incremented on failure, too?
+		// NOTE: nonce for creation from contracts starts at 1
+		// TODO: check if sender is an EOA and do not pre-increment
+		sender.nonce++;
+
+		auto encodeRlpInteger = [](int value) -> bytes {
+			if (value == 0) {
+				return bytes{128};
+			} else if (value <= 127) {
+				return bytes{static_cast<uint8_t>(value)};
+			} else if (value <= 0xff) {
+				return bytes{128 + 1, static_cast<uint8_t>(value)};
+			} else if (value <= 0xffff) {
+				return bytes{128 + 55 + 2, static_cast<uint8_t>(value >> 8), static_cast<uint8_t>(value)};
+			} else {
+				assertThrow(false, Exception, "Can only encode RLP numbers <= 0xffff");
+			}
+		};
+
+		bytes encodedNonce = encodeRlpInteger(sender.nonce);
+
 		h160 createAddress(keccak256(
+			bytes{static_cast<uint8_t>(0xc0 + 21 + encodedNonce.size())} +
+			bytes{0x94} +
 			bytes(begin(message.sender.bytes), end(message.sender.bytes)) +
-			asBytes(to_string(sender.nonce++))
-		));
+			encodedNonce
+		), h160::AlignRight);
+
 		message.destination = convertToEVMC(createAddress);
+		assertThrow(accounts.count(message.destination) == 0, Exception, "Account cannot exist");
+
 		code = evmc::bytes(message.input_data, message.input_data + message.input_size);
 	}
 	else if (message.kind == EVMC_CREATE2)
 	{
 		h160 createAddress(keccak256(
-			bytes(1, 0xff) +
+			bytes{0xff} +
```

Comment:
Is this correct? I think that `bytes(1, 0xff)` is a single-element vector containing `0xff` while `bytes{0xff}` is a vector of 255 zeros.

Reply:
Ah, wait. It's ok. It's an initializer list now.